src/Data/Fin/Relation/Unary/Top.agda

```diff
@@ -0,0 +1,139 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- The 'top' view of Fin
+--
+-- This is an example of a view of (elements of) a datatype,
+-- here i : Fin (suc n), which exhibits every such i as
+-- * either, i = fromℕ n
+-- * or, i = inject₁ j for a unique j : Fin n
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+module Data.Fin.Relation.Unary.Top where
+
+open import Data.Nat.Base using (ℕ; zero; suc; _<_)
+open import Data.Fin.Base using (Fin; zero; suc; toℕ; fromℕ; inject₁)
+open import Data.Fin.Properties using (toℕ-fromℕ; toℕ-inject₁; inject₁ℕ<)
+open import Relation.Binary.PropositionalEquality.Core
+open import Relation.Nullary.Negation using (contradiction)
+
+private
+  variable
+    n : ℕ
+    i : Fin (suc n)
+    j : Fin n
+
+------------------------------------------------------------------------
+-- The View, considered as a unary relation on Fin n
+
+-- NB `Data.Fin.Properties.fromℕ≢inject₁` establishes that the following
+-- inductively defined family on `Fin n` has constructors which target
+-- *disjoint* instances of the family (at index {suc n}); and hence the
+-- interpretations of the View constructors will also be disjoint
+
+data View : ∀ {n} (j : Fin n) → Set where
+  top : View (fromℕ n)
+  inj : {j : Fin n} → View j → View (inject₁ j)
+
+pattern inj₁ {n} j = inj {n = n} {j = j} _
```

Comment:
Does that cause trouble if you're importing both this module and `Data.Sum.Base`?
I think pattern desugaring is not type-directed so the overloading may be problematic.

Reply:
How about `fromℕ̌` and `inject₁̌ `  for the view's constructors?

Reply:
oh... hadn't considered such possibility (how do I write the backtick, much less quote it in Markdown?)...

... but (despite naming conventions elsewhere), I personally loathe `fromℕ` for the function (I've always found `top` to be more mnemonic...), and the choices of `inject₁` etc. in `Data.Fin.Base` are certainly... suboptimal in my view, so this addition was an opportunity to break with those choices. 

Your 'quoted' versions drag me back to them, but maybe you're correct to do so ... (but I find it *so* annoying that they are not the same length symbols, though... despite their particular arities)

Reply:
OK... I might be persuadable, subject to the qualification that I'd make ``inject₁` the derived form, and retain ``inj₁` for the view constructor? That way, the occurrence of a pattern of the form ``inject₁ j` really does correlate with a viewed expression of value `inject₁ j`...

... will sleep on it and commit tomorrow.

Reply:
That only leaves the question now of what to call the module itself: `Top` or `Fromℕ`, for consistency's sake?